### DISCUSSION TITLE
rbenv install 3.1.2 Build Failed checksum mismatch

### DISCUSSION THREAD

**Discussion by Sgtkeebs** [rbenv/ruby-build#2128] [Build failures] [answered]
Hello everyone,

I am going through the Odin Project ruby tutorial, and I am trying to use rbenv to install ruby version 3.1.2 but no matter what I do I always get a checksum mismatch error. I have tried following the tutorial on GitHub to no success. I figured that the files are becoming corrupt when I use git clone, so I manually downloaded them from the github page, but that didn't work. I tried installing rbenv with sudo apt install rbenv, and then copying the updated ruby-build files over but I am still getting the checksum mismatch. I am currently on WSL1 Ubuntu 22.04.

**Comment by mislav** [comment, marked as answer]
Hi, I'm sorry you're having trouble. I haven't heard of this problem before. Can you paste the exact output you are getting, plus these pieces of information:

1. Output of `rbenv install --version`
2. Output of `type shasum openssl sha256sum`

> I figured that the files are becoming corrupt when I use git clone

That is very unlikely. It should be safe to install ruby-build using git clone, and if anything was corrupted, there would be other errors than checksum mismatch.

**Reply by Sgtkeebs** [reply]
Thank you for your response and for your time, but may we please close out this discussion? I don't believe anything is wrong with Rbenv at this point because I tried to install homebrew to get around the mismatch, but homebrew is also giving me a mismatch error as well, and from checking the homebrew page and performing research it looks like it might be a WSL1 issue, so I am just going to be setting up a virtual machine instead. 

Edit: I don't know if it is the environment I am working in, but WSL1 did not work for me at all. I setup my virtual machine of ubuntu and I was able to successfully install HomeBrew as well as rbenv with ruby 3.1.2 successfully with no checksum mismatch error. I apologize for not trying this sooner!

**Reply by mislav** [reply]
No worries; I'm glad you solved it! Unlike GitHub issues, Discussions cannot be "closed", but I'll mark this thread as an answer.
